man/systemd.network.xml

```diff
@@ -2435,6 +2424,31 @@ IPv6Token=prefixstable:2002:da8:1::</programlisting></para>
         Defaults to <literal>yes</literal>.
         </listitem>
       </varlistentry>
+      <varlistentry>
+        <term><varname>RelayTarget=</varname></term>
+        <listitem>
+          <para>Takes an IPv4 address, which must be in the format
+          described in
+          <citerefentry project='man-pages'><refentrytitle>inet_pton</refentrytitle><manvolnum>3</manvolnum></citerefentry>.
+          Turns this DHCP server into a DHCP relay agent. See <ulink url="https://tools.ietf.org/html/rfc1542">RFC 1542</ulink>.
+          The address is the address of DHCP server or another relay agent to forward DHCP messages to and from.
+          Check also <varname>BindToInterface=</varname> option. Turning it off is required for relaying messages outside.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>RelayAgentCircuitId=</varname></term>
+        <listitem>
+          <para>Specifies value for Agent Circuit ID sub-option of Relay Agent Information Option. Takes a string.
+          Defaults to unset. Ignored if <varname>RelayTarget=</varname> is not specified.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>RelayAgentRemoteId=</varname></term>
+        <listitem>
+          <para>Specifies value for Agent Remote ID sub-option of Relay Agent Information Option. Takes a string.
+          Defaults to unset. Ignored if <varname>RelayTarget=</varname> is not specified.</para>
+        </listitem>
```

Comment:
So the assumption is that both CircuitId and RemoteId are fixed for the link. For *some cases* this is actually OK: e.g. a dhcp server started to give an address to a container... But in other cases, we'd actually want to include client info, for example our internal interface name (hashed?) as CircuitId, and the client MAC address as RemoteId… I think it is OK to merge this with just this single fixed value, but we should be prepared to allow more flexibility in the future (if there's ever a request for this).

Also, c-unescaping should be done because people might want to send non-printable strings? Please see how SendOption= is done… We have `config_parse_dhcp_send_option`, and I think we should reuse the same user interface and code internally as much as possible. @yuwata opinion?

I think this should have semantics similar to SendOption, e.g. `RelayAgentCircuitId=string:foobar`.

Reply:
Agree. Made a primitive SendOption= like format, which supports only "string:". This way there should be no backward compatibility problems when more advanced functionality will be needed